generator/internal/discovery_to_proto.cc

```diff
@@ -314,6 +314,28 @@ std::set<std::string> FindAllRefValues(nlohmann::json const& json) {
   return ref_values;
 }
 
+Status EstablishTypeDependencies(
+    std::map<std::string, DiscoveryTypeVertex>& types) {
+  for (auto& type : types) {
+    auto const& json = type.second.json();
+    auto ref_values = FindAllRefValues(json);
+    for (auto const& ref : ref_values) {
+      auto ref_iter = types.find(ref);
+      if (ref_iter == types.end()) {
+        return internal::InternalError(
+            "Unknown depended upon type",
+            GCP_ERROR_INFO()
+                .WithMetadata("dependent type", type.first)
+                .WithMetadata("depended upon type", ref));
```

Comment:
I would suggest that the name of the unknown, depended-upon type is more than metadata, and should be included in the message as well (instead?).